zstd/bytebuf.go

```diff
@@ -54,7 +54,7 @@ func (b *byteBuf) readBig(n int, dst []byte) ([]byte, error) {
 func (b *byteBuf) readByte() (byte, error) {
 	bb := *b
 	if len(bb) < 1 {
-		return 0, nil
+		return 0, io.ErrUnexpectedEOF
```

Comment:
Please document io.ErrUnExpectedEOF is normal behavior behavior like `dat, err := dec.DecodeAll(...)`.

the `dat` is valid if `len(dat) != 0`, but this kind of manner looks like obey the go convention.

Reply:
well, even `n, err := conn.Write(...)` does like this.
maybe improving the comment is enough.

Reply:
but in my case, `dec.DecodeAll(buf, nil)`, it's valid zstd compressed data, why not return io.EOF instead of io.UnExpectedEOF?